pkg/cmd/server/kubernetes/master/master_config_test.go

```diff
@@ -31,51 +31,86 @@ func TestAPIServerDefaults(t *testing.T) {
+			DeleteCollectionWorkers: 1,
+			EnableGarbageCollection: true,
+			EnableWatchCache:        true,
+		},
+		SecureServing: &apiserveroptions.SecureServingOptions{
+			ServingOptions: apiserveroptions.ServingOptions{
+				BindAddress: net.ParseIP("0.0.0.0"),
+				BindPort:    6443,
 			},
-			DefaultStorageMediaType:                  "application/json",
-			AdmissionControl:                         "AlwaysAdmit",
-			AuthorizationMode:                        "AlwaysAllow",
-			DeleteCollectionWorkers:                  1,
-			MasterServiceNamespace:                   "default",
-			AuthorizationWebhookCacheAuthorizedTTL:   5 * time.Minute,
-			AuthorizationWebhookCacheUnauthorizedTTL: 30 * time.Second,
+			ServerCert: apiserveroptions.GeneratableKeyCert{
+				CertDirectory: "/var/run/kubernetes",
+				PairName:      "apiserver",
+			},
+		},
+		InsecureServing: &apiserveroptions.ServingOptions{
+			BindAddress: net.ParseIP("127.0.0.1"),
+			BindPort:    8080,
 		},
 		EventTTL: 1 * time.Hour,
 		KubeletConfig: kubeletclient.KubeletClientConfig{
-			Port: 10250,
+			Port:         10250,
+			ReadOnlyPort: 10255,
 			PreferredAddressTypes: []string{
 				string(apiv1.NodeHostName),
+				string(apiv1.NodeInternalDNS),
 				string(apiv1.NodeInternalIP),
+				string(apiv1.NodeExternalDNS),
 				string(apiv1.NodeExternalIP),
 				string(apiv1.NodeLegacyHostIP),
 			},
 			EnableHttps: true,
 			HTTPTimeout: time.Duration(5) * time.Second,
 		},
-		WebhookTokenAuthnCacheTTL: 2 * time.Minute,
+		Audit: &apiserveroptions.AuditLogOptions{},
+		Features: &apiserveroptions.FeatureOptions{
+			EnableProfiling: true,
+		},
+		Authentication: &kubeoptions.BuiltInAuthenticationOptions{
+			Anonymous:       &kubeoptions.AnonymousAuthenticationOptions{Allow: true},
+			AnyToken:        &kubeoptions.AnyTokenAuthenticationOptions{},
+			BootstrapToken:  &kubeoptions.BootstrapTokenAuthenticationOptions{},
+			ClientCert:      &apiserveroptions.ClientCertAuthenticationOptions{},
+			Keystone:        &kubeoptions.KeystoneAuthenticationOptions{},
+			OIDC:            &kubeoptions.OIDCAuthenticationOptions{},
+			PasswordFile:    &kubeoptions.PasswordFileAuthenticationOptions{},
+			RequestHeader:   &apiserveroptions.RequestHeaderAuthenticationOptions{},
+			ServiceAccounts: &kubeoptions.ServiceAccountAuthenticationOptions{},
+			TokenFile:       &kubeoptions.TokenFileAuthenticationOptions{},
+			WebHook:         &kubeoptions.WebHookAuthenticationOptions{CacheTTL: 2 * time.Minute},
+		},
+		Authorization: &kubeoptions.BuiltInAuthorizationOptions{
+			Mode: "AlwaysAllow",
+			WebhookCacheAuthorizedTTL:   5 * time.Minute,
+			WebhookCacheUnauthorizedTTL: 30 * time.Second,
+		},
+		CloudProvider: &kubeoptions.CloudProviderOptions{},
+		StorageSerialization: &kubeoptions.StorageSerializationOptions{
+			StorageVersions:        kapi.Registry.AllPreferredGroupVersions(),
+			DefaultStorageVersions: kapi.Registry.AllPreferredGroupVersions(),
```

Comment:
as a follow up, this would be a good place to actually enumerate the group versions and preferred group versions, so we get notified on rebases if the defaults have changed for kube (since we need to track that to make sure we're storing the same preferred versions so we can drop legacy groups at the same time kube does)